dpnp/random/_random.pyx

```diff
@@ -125,346 +124,427 @@ cpdef dparray dpnp_uniform(long low, long high, size, dtype=numpy.int32):
 
     return result
 
-
-def rand(d0, *dn):
+cdef class RandomState:
     """
-    Create an array of the given shape and populate it
-    with random samples from a uniform distribution over [0, 1).
-
-    Parameters
-    ----------
-    d0, d1, …, dn : The dimensions of the returned array, must be non-negative.
-
-    Returns
-    -------
-    out : Random values.
-
-    See Also
-    --------
-    random
-
+    TODO:
+    description
     """
+    def __init__(self, seed=None):
+        # TODO:
+        # self.bit_generator = None ?
+        cdef size_t seed_
```

Comment:
Why do you need this variable stored permanently in the class? You are keeping `engine` already.

Reply:
need to refactor, forgot to remove it.